src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,159 @@
+import classNames from 'classnames';
+import React, { KeyboardEventHandler, useCallback } from 'react';
+import { Todo } from '../../types/Todo';
+import '../../styles/index.scss';
+
+type Props = {
+  todos: Todo[];
+  newTodo: Todo | null;
+  isTodoOnLoad: boolean;
+  onDelete: (id: number) => void;
+  pickedTodoId: number[];
+  handleTodoToggle: (id: number, completed: boolean) => void;
+  handleBlur: () => void;
+  setTodoIdToChange: (num: number | null) => void;
+  todoIdToChange: number | null;
+  newTitle: string;
+  setNewTitle: (str: string) => void;
+  handleOnKeyDown: KeyboardEventHandler<HTMLInputElement>;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  newTodo,
+  isTodoOnLoad,
+  onDelete,
+  pickedTodoId,
+  handleTodoToggle,
+  handleBlur,
+  setTodoIdToChange,
+  todoIdToChange,
+  newTitle,
+  setNewTitle,
+  handleOnKeyDown,
+}) => {
+  const callbackRef = useCallback((inputElement: HTMLInputElement) => {
+    if (inputElement) {
+      inputElement.focus();
+    }
+  }, []);
+
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {(todos.length > 0 || isTodoOnLoad) && (
+        todos.map((todo: Todo) => {
+          return (
+            <div
+              data-cy="Todo"
+              className={classNames(
+                'todo',
+                { completed: todo.completed },
+              )}
+              key={todo.id}
+            >
+              <label className="todo__status-label">
+                <input
+                  data-cy="TodoStatus"
+                  type="checkbox"
+                  className="todo__status"
+                  onClick={() => handleTodoToggle(todo.id, todo.completed)}
+                />
+              </label>
+
+              {todoIdToChange === todo.id ? (
+                <form
+                  onBlur={handleBlur}
+                >
+                  <input
+                    data-cy="TodoTitleField"
+                    type="text"
+                    className="todo__title-field"
+                    placeholder="Empty todo will be deleted"
+                    value={newTitle}
+                    onChange={(event) => {
+                      setNewTitle(event.target.value);
+                    }}
+                    ref={callbackRef}
+                    onKeyDown={handleOnKeyDown}
+                  />
+                </form>
+              ) : (
+                <>
+                  <span
+                    data-cy="TodoTitle"
+                    className="todo__title"
+                    onDoubleClick={() => {
+                      setTodoIdToChange(todo.id);
+                      setNewTitle(todo.title);
+                    }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)

Reply:
In general, as you wrote, it is good
As you want to write use [data-attributes](https://www.pluralsight.com/guides/how-to-access-custom-attributes-from-aevent-object-in-react)
![image](https://user-images.githubusercontent.com/70009205/205964193-b4359be8-9ad2-49a4-b6b7-ac64720cbc76.png)